# Integrative Bioinformatics (BIOINF 4220 - 3 ECTS)
## Overview
Many biological research questions are centered around trying to understand how small changes in
the organisms genome and environment can result in major changes in the cellular and organismal
phenotype. Examples for such changes in phenotype can be different body height/weight,
pathogenisis, or altered metabolic rates. To gain insights into the mechanism regulating how an
organism functions, ideally the full underlying biological system needs to be understood and
modeled. To do so, available measurements of the different layers in the central dogma
(transcription, translation) and further cellular function (metabolism) have to be integrated.

This course thus deals with one of the central tasks of bioinformatics: integration and unification of
biological data from different sources. Specifically, we well work with data sets for genome,
transcriptome, proteome and phenome measurements and try to coax out mechanistic insights into
the correlation between the genome and downstream processes (altered gene expression and altered
metabolism).

## Goals
* Data Integration using Python (and relevant packages)
* Genotype-Phenotype centered analyses (GWAS & PheWAS)
* Differential gene expression and pathway analysis
* Machine Learning with biological data

## Requirements
* Master in Bioinformatics (or at least the lectures Sequence Bioinformatics and Structure and Systems Bioinformatics)
* Proficient in a scripting language, preferably Python and/or willingness to pick the necessary skills prior (!!!) to the course

## Evaluation
* To pass the practical course a report will have to be handed in. The grade is based on this report and your performance during the course. Passing requires an overall grade of 4.0 or better.

## Preparation
* By starting the assignment, you have created a fork of the assignment repository, for you to work on your own.
* [Clone](https://education.github.com/git-cheat-sheet-education.pdf) this forked repo on your machine locally.
* [Install] conda on your machine(https://docs.anaconda.com/miniconda/install/)
* Open the terminal and navigate to the local directory.
* Create a conda environment and install the required packages. You can do this in one step by typing `conda env create -f environment.yml`.
* Active the conda environment with `conda activate pibi25`.
* Open jupyter lab with `jupyter lab`.
